### PR TITLE
Autotest add graph backward

### PR DESCRIPTION
**Pull request by BBuf**
- 给oneflow.nn.Module api构造的graph添加backward运行功能，当autotest打开`auto_backward=True`时（默认就是打开的），不仅会跑eager的backward测试（这里会对梯度做比较），还会跑对应graph的backward（这里不做梯度比较）。
- 修复global_atol错误拼写为gloal_atol的bug。
- 修复之前代码合并时eager passed这句信息的错位。
- 本地开启graph的单测可以正常通过。

### CODE REVIEW

File: python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py [hunk 1]
```diff
@@ -323,20 +325,46 @@ def dual_method(self, *args, **kwargs):
                                 ignore_apis_list = ["tensor", "train"]
                                 test_g_res = []
                                 if isinstance(oneflow, flow.nn.Module):
+                                    graph_train_args = []
+                                    for arg in oneflow_args:
```
Comment (BBuf) [inline python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py:329]: deepcopy的原因是这里使用eager module和eager tensor的副本，这样才不会修改module parameter以及eager tensor对应的梯度，即和eager的backward进行隔离。

**Review by strint — APPROVED** [review]
look good to me

**Review by MARD1NO — APPROVED** [review]

**Comment by strint** [comment]
CI抱怨的错误：
```
Error: onda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] [ERROR](GRAPH:TestGraphOfModule_3:TestGraphOfModule) building graph got error: <class 'oneflow._oneflow_internal.exception.CheckFailedException'> 
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter_util.cpp", line 139, in Dispatch<oneflow::one::Tensor>
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Dispatch<TensorTuple>(op_expr, inputs, ctx)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter_util.cpp", line 131, in Dispatch<oneflow::one::TensorTuple>
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Dispatch(op_expr, inputs, outputs.get(), ctx)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter.cpp", line 96, in Apply
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     internal_->Apply(op_expr, inputs, outputs, ctx)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/lazy_op_interpreter.cpp", line 634, in ApplyImpl
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Check failed: device_tag == GetDeviceTagOfTensor(input_tensor)  Lazy nn.Graph name : TestGraphOfModule_3 encountered ERROR where multi-input tensor has different device type in module/op_name: test_module-layer_norm-0. Please use tensor.to() or tensor.to_consistent() to make all input with same device.
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ERROR
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] 
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ======================================================================
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ERROR: test_layernorm_with_random_data_shared_mem (__main__.TestLayerNorm)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ----------------------------------------------------------------------
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] Traceback (most recent call last):
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py", line 761, in new_f
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     res = f(test_case)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py", line 182, in test_layernorm_with_random_data_shared_mem
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     y = m(x)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py", line 368, in dual_method
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     test_g_res = test_g(*oneflow_args)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/graph.py", line 258, in __call__
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     self._compile(*args)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/graph.py", line 506, in _compile
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     eager_outputs = self._build_graph(*args)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/graph.py", line 598, in _build_graph
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     outputs = self.build(*lazy_args)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py", line 354, in build
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     res = self.test_module(*args)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/block.py", line 209, in __call__
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     result = self.__block_forward(*args)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/block.py", line 241, in __block_forward
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     result = self._origin.__class__.forward(self, *args)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/modules/normalization.py", line 304, in forward
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     epsilon=self.eps,
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] oneflow._oneflow_internal.exception.CheckFailedException: 
    per_gpu_process_num=per_gpu_process_num,
  File "/opt/conda/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter_util.cpp", line 139, in Dispatch<oneflow::one::Tensor>
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Dispatch<TensorTuple>(op_expr, inputs, ctx)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter_util.cpp", line 131, in Dispatch<oneflow::one::TensorTuple>
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Dispatch(op_expr, inputs, outputs.get(), ctx)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter.cpp", line 96, in Apply
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     internal_->Apply(op_expr, inputs, outputs, ctx)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/lazy_op_interpreter.cpp", line 634, in ApplyImpl
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Check failed: device_tag == GetDeviceTagOfTensor(input_tensor)  Lazy nn.Graph name : TestGraphOfModule_3 encountered ERROR where multi-input tensor has different device type in module/op_name: test_module-layer_norm-0. Please use tensor.to() or tensor.to_consistent() to make all input with same device.
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] 
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ----------------------------------------------------------------------
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] Ran 2 tests in 14.706s
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] 
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] FAILED (errors=1)
[/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] 
    return future.result()
  File "/home/ci-user/runners/gpu/_work/oneflow/oneflow/ci/test/parallel_run.py", line 106, in run_cmds
    for cmd_gpu_num in everyN(cmdN, gpu_num)
  File "/home/ci-user/runners/gpu/_work/oneflow/oneflow/ci/test/parallel_run.py", line 88, in spawn_shell_and_check
    raise RuntimeError(cmd)
RuntimeError: /opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose
Error: Process completed with exit code 1.
```

**Comment by BBuf** [comment]
> CI抱怨的错误：
> 
> ```
> Error: onda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] [ERROR](GRAPH:TestGraphOfModule_3:TestGraphOfModule) building graph got error: <class 'oneflow._oneflow_internal.exception.CheckFailedException'> 
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter_util.cpp", line 139, in Dispatch<oneflow::one::Tensor>
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Dispatch<TensorTuple>(op_expr, inputs, ctx)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter_util.cpp", line 131, in Dispatch<oneflow::one::TensorTuple>
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Dispatch(op_expr, inputs, outputs.get(), ctx)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter.cpp", line 96, in Apply
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     internal_->Apply(op_expr, inputs, outputs, ctx)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/lazy_op_interpreter.cpp", line 634, in ApplyImpl
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Check failed: device_tag == GetDeviceTagOfTensor(input_tensor)  Lazy nn.Graph name : TestGraphOfModule_3 encountered ERROR where multi-input tensor has different device type in module/op_name: test_module-layer_norm-0. Please use tensor.to() or tensor.to_consistent() to make all input with same device.
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ERROR
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] 
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ======================================================================
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ERROR: test_layernorm_with_random_data_shared_mem (__main__.TestLayerNorm)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ----------------------------------------------------------------------
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] Traceback (most recent call last):
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py", line 761, in new_f
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     res = f(test_case)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py", line 182, in test_layernorm_with_random_data_shared_mem
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     y = m(x)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py", line 368, in dual_method
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     test_g_res = test_g(*oneflow_args)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/graph.py", line 258, in __call__
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     self._compile(*args)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/graph.py", line 506, in _compile
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     eager_outputs = self._build_graph(*args)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/graph.py", line 598, in _build_graph
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     outputs = self.build(*lazy_args)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py", line 354, in build
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     res = self.test_module(*args)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/block.py", line 209, in __call__
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     result = self.__block_forward(*args)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/graph/block.py", line 241, in __block_forward
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     result = self._origin.__class__.forward(self, *args)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/opt/conda/lib/python3.7/site-packages/oneflow/nn/modules/normalization.py", line 304, in forward
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     epsilon=self.eps,
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] oneflow._oneflow_internal.exception.CheckFailedException: 
>     per_gpu_process_num=per_gpu_process_num,
>   File "/opt/conda/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter_util.cpp", line 139, in Dispatch<oneflow::one::Tensor>
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Dispatch<TensorTuple>(op_expr, inputs, ctx)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter_util.cpp", line 131, in Dispatch<oneflow::one::TensorTuple>
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Dispatch(op_expr, inputs, outputs.get(), ctx)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/op_interpreter.cpp", line 96, in Apply
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     internal_->Apply(op_expr, inputs, outputs, ctx)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/framework/op_interpreter/lazy_op_interpreter.cpp", line 634, in ApplyImpl
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose]     Check failed: device_tag == GetDeviceTagOfTensor(input_tensor)  Lazy nn.Graph name : TestGraphOfModule_3 encountered ERROR where multi-input tensor has different device type in module/op_name: test_module-layer_norm-0. Please use tensor.to() or tensor.to_consistent() to make all input with same device.
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] 
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] ----------------------------------------------------------------------
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] Ran 2 tests in 14.706s
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] 
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] FAILED (errors=1)
> [/opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose] 
>     return future.result()
>   File "/home/ci-user/runners/gpu/_work/oneflow/oneflow/ci/test/parallel_run.py", line 106, in run_cmds
>     for cmd_gpu_num in everyN(cmdN, gpu_num)
>   File "/home/ci-user/runners/gpu/_work/oneflow/oneflow/ci/test/parallel_run.py", line 88, in spawn_shell_and_check
>     raise RuntimeError(cmd)
> RuntimeError: /opt/conda/bin/python3 /home/ci-user/runners/gpu/_work/oneflow/oneflow/test_tmp_dir/modules/test_layernorm.py --failfast --verbose
> Error: Process completed with exit code 1.
> ```
> 
> 对应这个case：test_layernorm_with_random_data_shared_mem interpreter里面提示该op的多个输入的tensor的device不一样

我的代码写错了，module拷贝之后忘记to device了，已解决。

File: python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py [hunk 2]
```diff
@@ -337,14 +341,39 @@ def dual_method(self, *args, **kwargs):
                                 ignore_apis_list = ["tensor", "train"]
                                 test_g_res = []
                                 if isinstance(oneflow, flow.nn.Module):
+                                    graph_train_oneflow = copy.deepcopy(oneflow).to(
```
Comment (hjchen2) [inline python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py:344]: 这个to(arg_device_type)操作能去掉么，因为consistent tensor没有device

File: python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py [hunk 3]
```diff
@@ -319,9 +321,11 @@ def dual_method(self, *args, **kwargs):
                             oneflow_res = torch_tensor_to_flow(pytorch_res)
                         else:
                             graph_args = []
+                            arg_device_type = "cpu"
```
Comment (hjchen2) [inline python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py:324]: 从这里开始到338行，是不是都应该放到`if testing_graph:`分支里面去

File: python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py (hunk 2 above)
Comment (BBuf) [reply python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py:344]: 这个功能是依赖这个，要不直接用consistent那个环境变量在这里做个判断吧？如果是consistent就skip

Comment (hjchen2) [reply python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py:344]: 好，那我想办法改一下吧

File: python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py (hunk 3 above)
Comment (BBuf) [reply python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py:324]: 感觉是，我后面改下。

**Pull request merged**